ibis/backends/datafusion/__init__.py

```diff
@@ -210,7 +210,8 @@ def read_csv(
             The just-registered table
         """
         path = normalize_filename(path)
-        table_name = table_name or gen_name("read_csv")
+        # Datafusion requires lowercase table names
+        table_name = table_name or gen_name("read_csv").lower()
```

Comment:
I don't think we should do this on behalf of the user. The user needs to be aware of this lack of functionality in datafusion so that when they name a table `myTable`, it's not magically named something different because of missing functionality in datafusion. I don't even think we should check for this and raise an error, let the datafusion error bubble up.

This change also seems unrelated to the goal of the PR (at least from the title anyway).

Reply:
This is just a leftover after a rebase.